### PR TITLE
chore: bump revm+alloy

### PR DESCRIPTION
**Pull request by mattsse**
blocked by https://github.com/alloy-rs/trie/pull/8

this will unblock more simplifications:
* rm reth rpc admin types
* use alloy for testing

cc @AbnerZheng 

receipt handling is now a bit more verbose but this can be simplified with followups on alloy

### CODE REVIEW

File: crates/rpc/rpc/src/otterscan.rs [hunk 1]
```diff
@@ -155,10 +158,48 @@ where
         }
 
         // Crop receipts and transform them into OtsTransactionReceipt
-        let timestamp = u64::try_from(block.header.timestamp).unwrap_or(u64::MAX);
+        let timestamp = Some(u64::try_from(block.header.timestamp).unwrap_or(u64::MAX));
         let receipts = receipts
             .drain(page_start..page_end)
-            .map(|receipt| OtsTransactionReceipt { receipt, timestamp })
+            .map(|receipt| {
+                let TransactionReceipt {
```
Comment (onbjerg) [inline crates/rpc/rpc/src/otterscan.rs:165]: we should really upstream some conversion helpers for this lol

**Review by onbjerg — APPROVED** [review]
slowly but surely 🫡

File: crates/rpc/rpc/src/otterscan.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc/rpc/src/otterscan.rs:165]: https://github.com/alloy-rs/alloy/pull/460

Comment (onbjerg) [reply crates/rpc/rpc/src/otterscan.rs:165]: beast mode

**Pull request merged**